### DISCUSSION TITLE
`StatusContext` interface or testing?

### DISCUSSION THREAD

**Discussion by mlapaglia** [spectreconsole/spectre.console#1538] [Q&A] [answered]
is passing `StatusContext` into helper methods recommended? i am having issues coming up with a way to write tests for this
```
private async Task<Stream> DownloadStuffAsync(
    StatusContext ctx,
    CancellationToken cancellationToken)
{
    ctx.Status("Searching for file to download...");
```
since there isn't an `IStatusContext` interface i am assuming i'm not following a good practice. ideas?

**Comment by phil-scott-78** [comment]
I'd pass in a `Action<string>` rather than coupling that method to UI concerns. Another option would be wrapping the `StatusContext` if you truly need to pass it around

**Reply by patriksvensson** [reply, marked as answer]
Agree, but the way it's really meant to be tested, is using the TestConsole in Spectre.Console.Testing, and run your app as you normally do.
